packages/core/src/ceramic.ts

```diff
@@ -312,7 +313,7 @@ class Ceramic implements CeramicApi {
 
     const doPeerDiscovery = config.useCentralizedPeerDiscovery ?? !TESTING
     if (doPeerDiscovery) {
-      await periodicallyConnectToPeers(networkOptions.name, ipfs)
+      await topology.start()
```

Comment:
Instead of doing this here, since this functionality also exists in IpfsDaemon, can we just make sure to pass the proper value for `useCentralizedPeerDiscovery` when we construct the IpfsDaemon in `build-ipfs-connection.ts`?

Reply:
I am worried about one scenario: dummy (=no knowledge of Ceramic network) IPFS node and Ceramic core instance running atop. In the scenario, if `topology.start` is removed here, the connection will never be established to the network.

IPFS Daemon can be run as an independent node with CERAMIC_NETWORK passed in, so that the daemon establishes connection to other IPFS nodes. Also, it can be run without CERAMIC_NETWORK passed in. Then the daemon has no idea what to connect to. Ceramic core instance though always knows a network, so it can freely command a dummy IPFS daemon to connect to proper peers.

Ceramic core instance commands peering topology for IPFS node - that's the main scenario, even more valid for go/rust IPFS nodes. IPFS daemon making own topology based on network is a nice benefit, which is valuable from deployment perspective.

Reply:
Yeah, that makes sense, good call.  I'd probably prefer to put this logic into `buildIpfsConnection` so its contained within the rest of the ipfs-specific logic rather than up here in the Ceramic initializer, but not a huge deal either way.

Reply:
`buildIpfsConnection` is called _only_ when Ceramic daemon is started from cli. One _can_ (and it is useful) run ceramic instance from a Nodejs script against an unknown IPFS node. In this case, if peer-discovery is out of Ceramic instance codebase, no peer discovery is set up.

Reply:
If one considers a case of Ceramic node running against dummy IPFS node to be valid, peer-discovery has to reside in Ceramic instance.

I acknowledge, there is alternative way of doing things: expose network name from CeramicAPI, and let a consumer apply peer-discovery by herself. Not sure, though if it makes sense, as peer-discovery anyway has to be setup out of the box, be it for IPFS or Ceramic-Ceramic connection.

Reply:
oh whoops, sorry my mistake.  I thought we were using `buildIpfsConnection` within Ceramic.create() to get the ipfs connection handle, but I see it's actually passed in and expected to be already set up. So yeah, the way you're doing it here makes perfect sense.  Thanks for bearing with me.

Reply:
Resolving the conversation then :)